src/JWT/Builder/JwtBuilder.cs

```diff
@@ -468,5 +474,42 @@ private bool CanDecodeHeader()
 
             return true;
         }
+
+        private string GetPropName(MemberInfo prop)
+        {
+            var jsonSerializer = _jsonSerializerFactory.Create();
+
+            var customAttributes = prop.GetCustomAttributes(inherit: true);
+            foreach (var attribute in customAttributes)
+            {
+                switch (jsonSerializer)
+                {
+                    case JsonNetSerializer:
+                    {
+                        if (attribute is JsonPropertyAttribute jsonNetProperty)
+                        {
+                            return jsonNetProperty.PropertyName;
+                        }
+                        break;
+                    }
+#if MODERN_DOTNET
+                    case SystemTextSerializer:
+                    {
+                        if (attribute is JsonPropertyNameAttribute stjProperty)
+                        {
+                            return stjProperty.Name;
+                        }
+                        break;
+                    }
+#endif
+                    default:
+                    {
+                        throw new NotSupportedException($"{jsonSerializer.GetType()} is not supported");

```

Comment:
I don't think we should throw an exception here. It defeat the purpose of having an interface, we don't know if the customers have their own implementation, the idea is that they might.

Reply:
Can the `default` case be just omitted so the method would return `prop.Name`?

Reply:
good idea